actionpack/lib/action_dispatch/log_subscriber.rb

```diff
@@ -9,6 +9,14 @@ def redirect(event)
 
       info { "Redirected to #{payload[:location]}" }
 
+      if ActionDispatch.verbose_redirect_logs
+        request = payload[:request]
+
+        request.routes.router.recognize(request) do |route|
+          info { "↳ #{route.source_location}" }
+        end
+      end
+
```

Comment:
If the log level is higher than `:info`  we can skip some code execution by wrapping it all in the `info` block.
```suggestion
      info do
        if ActionDispatch.verbose_redirect_logs
          request = payload[:request]

          request.routes.router.recognize(request) do |route|
            "↳ #{route.source_location}"
            # Or maybe we need to assign it to a instance variable here:
            # message = "↳ #{route.source_location}"
          end
          # message # maybe?
        end
      end

```